lib/resin.coffee

```diff
@@ -219,3 +224,64 @@ module.exports = getSdk = (opts = {}) ->
 	sdk.errors = errors
 
 	return sdk
+
+###*
+# @summary Set shared default options
+# @name setSharedOptions
+# @public
+# @function
+# @memberof resin
+#
+# @description
+# Set options that are used by calls to `resin.fromSharedOptions()`.
+# The options accepted are the same as those used in the main SDK factory function.
+#
+# @params {Object} opts - The shared default options
+#
+# @example
+# resin.setSharedOptions({
+# 	apiUrl: 'https://api.resin.io/',
+# 	imageMakerUrl: 'https://img.resin.io/',
+# 	apiVersion: 'v2',
+# 	isBrowser: true,
+# });
+###
+getSdk.setSharedOptions = (opts) ->
+	defaults(opts, defaultOpts)
```

Comment:
I don't think you need to do this (since it'll get done within the eventual `getSdk` call anyway), and I think doing so might actually break cross-version compatibility (since in that case we'll be setting the defaults from the version of the library used to set the options, not the version of the library that contains the actual SDK code that's going to be used).